### PR TITLE
Cinema project using hibernate hw1

### PR DESCRIPTION
**Pull request by AndriiBardin**

### CODE REVIEW

File: src/main/java/cinema/dao/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    public Movie add(Movie movie) throws DataBaseException {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (Max-Pochepets) [inline src/main/java/cinema/dao/MovieDaoImpl.java]:
```suggestion
            session.save(movie);
            transaction.commit();
```
look [closely](https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html)

File: src/main/java/cinema/service/MovieService.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package cinema.service;
+
+import cinema.model.movie.Movie;
+import java.util.List;
+
+public interface MovieService {
+
```
Comment (Max-Pochepets) [inline src/main/java/cinema/service/MovieService.java]:
```suggestion
```

File: src/main/java/cinema/util/HibernateUtil.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package cinema.util;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+public class HibernateUtil {
+
```
Comment (Max-Pochepets) [inline src/main/java/cinema/util/HibernateUtil.java]:
```suggestion
```

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]
check if all your files have an empty line in the end

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/cinema/dao/MovieDao.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package cinema.dao;
+
+import cinema.exception.DataBaseException;
+import cinema.model.movie.Movie;
+import java.util.List;
+
+public interface MovieDao {
+    Movie add(Movie movie) throws DataBaseException;
```
Comment (Sofasmile) [inline src/main/java/cinema/dao/MovieDao.java]:
```suggestion
    Movie add(Movie movie);
```

File: src/main/java/cinema/dao/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) throws DataBaseException {
```
Comment (Sofasmile) [inline src/main/java/cinema/dao/MovieDaoImpl.java]:
```suggestion
    public Movie add(Movie movie) {
```

File: src/main/java/cinema/dao/MovieDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (DataBaseException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataBaseException("Can't add movie entity");
```
Comment (Sofasmile) [inline src/main/java/cinema/dao/MovieDaoImpl.java]:
```suggestion
            throw new DataBaseException("Can't add movie entity " + movie, e);
```

File: src/main/java/cinema/dao/MovieDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> getAllFilms = session.createQuery("from Movie", Movie.class);
+            return getAllFilms.getResultList();
+        } catch (DataBaseException e) {
+            throw new DataBaseException("Can't get all movies");
```
Comment (Sofasmile) [inline src/main/java/cinema/dao/MovieDaoImpl.java]:
```suggestion
            throw new DataBaseException("Can't get all movies", e);
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/dao/MovieDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (DataBaseException e) {
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/dao/MovieDaoImpl.java:24]:
```suggestion
        } catch (Exception e) {
```

File: src/main/java/cinema/dao/MovieDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> getAllFilms = session.createQuery("from Movie", Movie.class);
+            return getAllFilms.getResultList();
+        } catch (DataBaseException e) {
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/dao/MovieDaoImpl.java]: same

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**